spec/services/dttp/access_token_service_spec.rb

```diff
@@ -0,0 +1,45 @@
+require "rails_helper"
+
+module Dttp
+  describe AccessTokenService do
+    describe described_class::Client do
+      it "uses the microsoft base uri" do
+        expect(described_class.base_uri).to eq("https://login.microsoftonline.com")
```

Comment:
could re-use the constant if we put it into one here

Reply:
Ah right... that other benefit 🤣

Reply:
Actually, if I do that then we have no test that the correct URL is used. I'd have to add another test that the constant contains the correct URL. What do you think?

Reply:
Yeah that's true but it means that this test will fail if the url changes (for whatever reason) if that's intentional then we can leave it otherwise it'll just assert against whatever value is assigned to the constant. 

eg:

`expect(described_class.base_uri).to eq(AccessTokenService::Client::AUTH_URL)`

Reply:
But nothing will test the constant is set correctly. If I change it to a constant we could have the constant set to 'cbeebies.co.uk' and it would still pass. I get what you're saying but we'd need to test the constant value is correct if it's in the code. We could also configure it from settings but the same thing would apply there.